System.Security.Cryptography/AES.cs

```diff
@@ -0,0 +1,162 @@
+                EncryptBlock(key, lastBlock);
+                Array.Copy(lastBlock, 0, encryptedData, blockCount * blockSize, remainder);
+            }
+
+            return encryptedData;
+        }
+
+        /// <summary>
+        /// XOR the block of data with key
+        /// </summary>
+        /// <param name="key">The secret key for XOR opration with block </param>
+        /// <param name="block">The Block of data for XOR opration with secret key</param>
+        /// <exception cref="ArgumentException"></exception>
+        private void EncryptBlock(byte[] key, byte[] block)
+        {
+            // Ensure that the key and block have the same length
+            if (key.Length != block.Length)
+            {
+                throw new ArgumentException("Key and block must have the same length.");
+            }
+
+            for (int i = 0; i < block.Length; i++)
+            {
+                block[i] = (byte)(block[i] ^ key[i]);
+            }
+        }
+
+        /// <summary>
+        /// Decrypt the array of bytes In ECB Mode
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        private byte[] DecryptAesEcb(byte[] key, byte[] data)
+        {
+            int blockSize = 16; // AES block size is 128 bits (16 bytes)
+            int blockCount = data.Length / blockSize;
+            int remainder = data.Length % blockSize;
+
+            byte[] decryptedData = new byte[data.Length];
+
+            for (int i = 0; i < blockCount; i++)
+            {
+                byte[] block = new byte[blockSize];
+                Array.Copy(data, i * blockSize, block, 0, blockSize);
+                DecryptBlock(key, block);
+                Array.Copy(block, 0, decryptedData, i * blockSize, blockSize);
+            }
+
+            // If there is a remainder, pad the last block and decrypt
+            if (remainder > 0)
+            {
+                byte[] lastBlock = new byte[blockSize];
+                Array.Copy(data, blockCount * blockSize, lastBlock, 0, remainder);
+                DecryptBlock(key, lastBlock);
+                Array.Copy(lastBlock, 0, decryptedData, blockCount * blockSize, remainder);
+            }
+
+            return decryptedData;
+        }
+
+        /// <summary>
+        /// XOR the block of data with key
+        /// </summary>
+        /// <param name="key">The secret key for XOR opration with block </param>
+        /// <param name="block">The Block of data for XOR opration with secret key</param>
+        /// <exception cref="ArgumentException"></exception>
+        private void DecryptBlock(byte[] key, byte[] block)
+        {
+            // Ensure that the key and block have the same length
+            if (key.Length != block.Length)
+            {
+                throw new ArgumentException("Key and block must have the same length.");
```

Comment:
please move the text of the exception in the intellisense comments, so we save space in the device ;-)